source/mqtt/MqttClient.cpp

```diff
@@ -427,6 +431,15 @@ namespace Aws
             }
 
             bool MqttConnection::SetOnMessageHandler(OnPublishReceivedHandler &&onPublish) noexcept
+            {
+                return SetOnMessageHandler(
+                    [onPublish](
```

Comment:
Is the capture here correct?  Memory is fuzzy wrt capture movable type.

Reply:
Yeah, it just makes a copy of onPublish.

If we were using c++14, we could do a std::move when capturing, but alas